pkg/ebpf/tracee.go

```diff
@@ -408,17 +412,31 @@ func (t *Tracee) Init() error {
 		return fmt.Errorf("error creating process tree: %v", err)
 	}
 
-	// Initialize containers enrichment logic
+	// Initialize cgroups filesystems
 
-	capabilities.GetInstance().Requested(func() error { // TODO: workaround until PR: #2233 is in place
+	err = caps.Requested(func() error { // ring2
+		t.cgroups, err = cgroup.NewCgroups()
+		return err
+	}, cap.SYS_ADMIN)
+	if err != nil {
+		return err
```

Comment:
This will fail tracee from running. But is that what we really want? What if the user don't need cgroup programs?

Reply:
We need the mounting logic anyway when we run tracee inside a container (see from a while back #1338 #1900)

Reply:
But now it is always required, which makes the requirement for CAP_BPF and CAP_PERFMON on newer kernels irrelevant.
CAP_SYS_ADMIN is something we should avoid. We need to have the ability to let the user run tracee without requiring this capability - either by running the binary directly or by mounting the cgroup themselves.

Reply:
We only need to run this in a capabilities ring once during initialization and that's it, other than that we can keep running only with CAP_BPF and CAP_PERFMON.
I do think we should move this caps call to the cgroups method though, that way we can scope it only for the mount call itself.

Reply:
The thing is that there are environments where CAP_SYS_ADMIN won't be available for us, and we need to try to run tracee even without it (of course, if we can't continue - we can return an error)

Reply:
I'll do like what we've discussed and mount only when needed. For mounting, I'll raise to ring2 with CAP_SYSADMIN only (and will have to unmounted at the cleanup).

Reply:
This is being solved by new code.